### PR TITLE
KEYCLOAK-16227 - Add realm login settings to realm CRD

### PR DESCRIPTION
**Pull request by dhx-teuto**
## JIRA ID
KEYCLOAK-16227

Once this pull request is as you want it I can prepare the remaining ones in a similar way.

## Verification Steps

Add the steps required to check this change. Following an example.

1. Add some of changed options in the example realm
2. `kubectl apply` the realm
3. Open Admin Console
4. Check if the options are set correctly

## Checklist:
- [ ] Automated Tests
- [ ] [Documentation](https://github.com/keycloak/keycloak-documentation) changes if necessary
- [ ] (If required) Discussed on [Keycloak DEV](https://groups.google.com/forum/#!forum/keycloak-dev)

### CODE REVIEW

**Comment by slaskawi** [comment]
Dear contributor,

I'm putting this Pull Request on hold until we get our CI back (see https://github.com/keycloak/keycloak-operator/pull/266). Once the Github Actions Pull Request is merged, I will kindly ask you for rebasing your changes. Then, we'll proceed with a normal code review process.

Please see [this Keycloak DEV thread](https://groups.google.com/g/keycloak-dev/c/ijI9grpRw74) to learn more why we are migrating to Github Actions.

Sorry for inconvenience

**Comment by slaskawi** [comment]
@dhx-teuto Could you please rebase it?

**Review by miquelsi — APPROVED** [review]
LGTM. Thanks!

**Review by slaskawi — APPROVED** [review]
LGTM! Let's give others a day or two to look at this. Then, I'll merge it.

File: pkg/apis/keycloak/v1alpha1/keycloakrealm_types.go
Comment (vmuzikar) [inline pkg/apis/keycloak/v1alpha1/keycloakrealm_types.go:110]: Just a minor comment. Should this be a custom type instead of string? It's basically an enum. What's the best practise here @slaskawi?

File: pkg/apis/keycloak/v1alpha1/keycloakrealm_types.go
Comment (vmuzikar) [inline pkg/apis/keycloak/v1alpha1/keycloakrealm_types.go:82]: (A comment unrelated to this specific line.)

Should we add more realm properties like e.g. `clientSessionMaxLifespan`, `accessCodeLifespan`, etc. Should this be 1:1 with the realm representation provided by Keycloak?

**Review by vmuzikar — COMMENTED** [review]
@dhx-teuto Thank you for your contribution! Added two minor comments (more of a questions).

@slaskawi I've got a generic question, out of scope of this PR. Do we have some test mechanism that verifies all the settings in the CRD actually exist in Keycloak? I know we remove things from the representations at KC side rather rarely but it happens sometimes and unless manually checked we won't know it broke the operator. Another thing is when something is added at KC side. Do we want to always add it to the operator or it is on need basis?

File: pkg/apis/keycloak/v1alpha1/keycloakrealm_types.go
Comment (dhx-teuto) [reply pkg/apis/keycloak/v1alpha1/keycloakrealm_types.go:110]: My reasoning was to mimic https://www.keycloak.org/docs-api/11.0/rest-api/index.html#_realmrepresentation as closely as possible and while there are some places where enums are used this is not one of them.
If I should change it just let me know.

File: pkg/apis/keycloak/v1alpha1/keycloakrealm_types.go
Comment (dhx-teuto) [reply pkg/apis/keycloak/v1alpha1/keycloakrealm_types.go:82]: I have pull requests ready for email, theme, brute force. Just waiting for the rebase.
As for your question, it seems that it should https://issues.redhat.com/browse/KEYCLOAK-14321

Comment (vmuzikar) [reply pkg/apis/keycloak/v1alpha1/keycloakrealm_types.go:82]: Thanks! I didn't know there's a JIRA for this already. ;)

**Review by abstractj — APPROVED** [review]
LGTM

File: pkg/apis/keycloak/v1alpha1/keycloakrealm_types.go
Comment (slaskawi) [reply pkg/apis/keycloak/v1alpha1/keycloakrealm_types.go:110]: I'm fine with leaving this as String. The validation will be performed on Keycloak REST level.

**Comment by slaskawi** [comment]
Integrated, thanks @dhx-teuto !

**Pull request merged**
